### DISCUSSION TITLE
Seperate button for on and off

### DISCUSSION THREAD

**Discussion by cjs1502** [arendst/Tasmota#15459] [Q&A] [answered]
I have water tank which has two sensor,one at the top and one at the bottom.
I wanted to use those sensors to automate the filling of tank, i.e. when the water level goes below the lower sensor relay turns on and as soon as the water hits the upper sensor that same relay turns off.

In short i want two seperate buttons of which one only turns on the relay and other only turns off the relay regardless of high/low state.

I tried using buttons feature but when water goes down the lower sensor it turns on the relay and as soon as a little bit of water is added it senses again and turns of the relay.

I hope this gives a hint what i am trying to say

**Comment by barbudor** [comment, marked as answer]
Define as switches, decouple from relay and make a rule
See https://tasmota.github.io/docs/Buttons-and-Switches/#detach-switches-with and https://tasmota.github.io/docs/Rules/

**Comment by cjs1502** [comment]
This is my first ever tasmota device and i this a bit more overwhelming.
I couldn't understand a single thing written there 😅,it was too much complicated for me.
I tried espeasy and that was way complicated to use.
Is there any easy way i can do it?

**Reply by barbudor** [reply]
> Is there any easy way i can do it?

It's easy with Tasmota and Rules
But you need to learn how to do it

Assuming:
- Your sensors are connected to 2 GPIOs, configure those GPIO as `Switch 1` (bottom of tank) and `Switch 2` (top of tank)
- The sensors are working in a way that Tasmota see OFF when there is no liquid at that sensor and ON when there is liquid
- There is a Relay connected to a GPIO configured as `Relay 1`

Decouple the switches from the Relay
``` haskell
backlog SwitchMode1 15; SwitchMode2 15
```
Create the following rule:
``` haskell
Rule1 ON Switch1#State==0 DO Power1 ON ENDON ON Switch2#State==1 DO Power1 OFF ENDON
Rule1 1
Rule1 5
```

**Reply by cjs1502** [reply]
Thank you for this detailed reply,i will try this.
